# Functional Analysis Based on Genome-wide Association Study Meta-analysis in Abdominal Aortic Aneurysm 

Abdominal aortic aneurysm (AAA) remains one of the most fatal non-communicable diseases worldwide, despite advances in surgical and therapeutic approaches. Genetics and molecular mechanisms are known to play crucial roles in the development and progression of various diseases, including infectious diseases, cancers, cardiovascular diseases, and neurodegenerative disorders.

Moreover, since the genomic and postgenomic eras, bioinformatics and omics technologies have become crucial components of biomedical science. Therefore, we implemented integrative bioinformatics pipelines to investigate the molecular mechanisms underlying AAA. First, we conducted a GWAS meta-analysis based on summary statistics of AAA obtained from the GWAS Catalog (https://www.ebi.ac.uk/gwas/), using METAL ([https://genome.sph.umich.edu/wiki/METAL_Quick_Start](https://genome.sph.umich.edu/wiki/METAL_Documentation)) for the meta-analysis. Subsequently, we identified functional pathways and performed expression quantitative trait loci (eQTL) analysis using FUMA (https://fuma.ctglab.nl/).

We then identified differentially expressed genes (DEGs) from two sources: eQTLs identified through FUMA and significant genes from a transcriptome-wide association study (TWAS) analysis, performed using FUSION (http://gusevlab.org/projects/fusion/). The DEGs were used to construct an AAA-related protein–protein interaction (PPI) network based on human interactome data from the STRING database (https://string-db.org/), with visualization carried out using Cytoscape. Key nodes were identified using NetworkX in Python, and drug repurposing was performed to find FDA-approved drug candidates targeting the key proteins through searches of several drug–gene and drug–protein interaction databases. We validated the drug–protein interaction results through molecular docking using the HDOCK server (http://hdock.phys.hust.edu.cn/).

To support transparency and reproducibility, we created a GitHub repository to collect the source code used in our data analysis. For detailed methodologies and results, please refer to our manuscript.
